python3/Makefile

```diff
@@ -1,9 +1,160 @@
+  # like renaming a few files after after the install to remove their .py extension
+  #
+
+  install ::
+>   @# If we found files that matched the wildcard and weren't excluded by the filter
+>   @# then install them in the target directory $(DESTDIR)/$(3)
+    ifneq ($$($(2)),)
+>     @# Create the target directory if it doesn't exist
+>     @mkdir -p                      $(DESTDIR)$(3);set -xv
+      # Install the files in the target directory using the given IPRPG or IDATA macro
+>     @(cd $(2); $$($(1)) -v $$($(2)) $(DESTDIR)$(3))
+      # Call the optional macro in  to make adjustments to the installed files if given
+      ifneq ($(4),)
+>       @$(call $(4), $$($(2)), $(DESTDIR)$(3))
+      endif
+  endif
+endef
+# Macro to remove the .py extension from the given programs in the install directory
+# The 1st argument is the list of programs
+# The 2nd argument is the installation directory
+define REMOVE_DOT_PY
+  cd $(2); for f in $(1); do mv -v $$$$f $$$${f%.py}; done
+endef
+# -----------------------------------------------------------------------------
+# When testing, the first part of the install target is to  cleanup the temp dir
+TESTDESTDIR = /tmp/$(USER)-xen-api-python3
+ifeq ($(DESTDIR),$(TESTDESTDIR))
+# Example test command (for editing in another terminal while the test is running):
+# while inotifywait python3/Makefile;do make -C --no-print-directory python3;done
+install::
+# If install target is the default temporary target, cleanup the temporary directory
+> rm -rf $(DESTDIR)
+endif
+
+# -----------------------------------------------------------------------------
+# Define the install rule for the packages and programs:
+# -----------------------------------------------------------------------------
+# The programs are installed in the $OPTDIR/bin directory
+# The .py extension is removed from the programs
+$(eval $(call INSTALL_FILES,IPROG,bin,$(OPTDIR)/bin,REMOVE_DOT_PY))
+
+# The packages are installed in the python site-packages directory
+$(eval $(call INSTALL_FILES,IDATA,packages,$(SITE3_DIR)))
+# The libraries are installed in the libexec directory
+$(eval $(call INSTALL_FILES,IPROG,libexec,$(LIBEXECDIR)))
+# The plugins are installed in the PLUGINDIR directory
+$(eval $(call INSTALL_FILES,IPROG,plugins,$(PLUGINDIR)))
+
+# The yum-plugin files are installed in the PLUGINDIR directory
+$(eval $(call INSTALL_FILES,IPROG,yum-plugins,$(YUMPLUGINDIR)))
+# The yum-plugin config files are installed in the PLUGINCONFDIR directory
+$(eval $(call INSTALL_FILES,IPROG,yum-plugins/configs,$(YUMPLUGINCONFDIR)))
+
+# The extensions are installed in the EXTENSIONDIR directory
+$(eval $(call INSTALL_FILES,IPROG,extensions,$(EXTENSIONDIR),REMOVE_DOT_PY))
 
-install:
-	mkdir -p $(DESTDIR)$(SITE3_DIR)
-	$(IDATA) packages/observer.py $(DESTDIR)$(SITE3_DIR)/
+# The poweron plugins are installed in the PLUGINDIR directory
+# TODO: Some need to be renamed in the target directory to remove
```

Comment:
I'd rather have known manual, mechanical work that needs to be done per file rather than having an automated tool with todos.

Reply:
Removed.